spec/enumerable_methods_spec.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+require_relative '../enumerable_methods'
+
+describe '#my_each' do
+  let(:array) { [1, 2, 3, 4, 5] }
+
+  it 'with a block' do
+    expect(array.my_each { |i| i * i }).to eql([1, 4, 9, 16, 25])
+  end
+  it 'without a block' do
+    expect(array.my_each).to be_an(Enumerator)
+  end
+end
+
+describe '#my_each_with_index' do
```

Comment:
This behaves like ```.each``` with extra functionality of returning the index of current element. 
I am not too sure what this test seeks to do. I don't think its being equal to nil has any bearing on the functionality of the method. maybe we can keep it simple.